Mathlib/Algebra/Associated.lean

```diff
@@ -313,44 +316,36 @@ theorem IsSquare.not_irreducible (ha : IsSquare a) : ¬Irreducible a := fun h =>
 
 end CommMonoid
 
+section CommMonoidWithZero
+
+variable [CommMonoidWithZero α] [DecompositionMonoid α] {a : α} (irr : Irreducible a)
+
+theorem Irreducible.prime_of_isPrimal (primal : IsPrimal a) : Prime a :=
```

Comment:
I suppose that Lean is smart enough to not include the `DecompositionMonoid` instance in this statement? But I would like to have a bit more of a formal guarantee, even if that means duplicating a bit of code...